versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractManyKeys.java

```diff
@@ -524,4 +524,56 @@ private static void checkKeysAndValuesIndividually(
           .isEqualTo(valueGen.apply(i).serialized());
     }
   }
+
+  public static Stream<List<String>> progressivelyManyKeyNames() {
+    // The seed of `7` was chosen after a few manual tries because it caused the most test failures
+    // in the key lookup code before the commit where this test was introduced.
+    Random random = new Random(7);
+    return IntStream.range(1, 100)
+        // Use two test args of the same size (with random, deterministic values)
+        .flatMap(i -> IntStream.of(i, i))
+        .mapToObj(
+            i ->
+                IntStream.range(1, i)
+                    .mapToObj(j -> Long.toString(Math.abs(random.nextLong()), Character.MAX_RADIX))
+                    .collect(Collectors.toList()));
+  }
+
+  /**
+   * Validates key construction and lookup with random, yet deterministic key names hoping to cover
+   * all edge cases and hash collisions by using a somewhat large test data set.
+   *
+   * <p>Note: the number of keys that is effective for validating the hash lookup indirectly depends
```

Comment:
Hint: You can get a `DatabaseAdapter` instance here as a method parameter, using the `@NessieDbAdapter` plus two `@NessieDbAdapterConfigItem` annotations for the method parameter to ensure the test's using the correct adapter configuration.

Reply:
updated